gmso/formats/lammpsdata.py

```diff
@@ -225,3 +235,265 @@ def write_lammpsdata(topology, filename, atom_style='full'):
                 topology.sites.index(dihedral.connection_members[2])+1,
                 topology.sites.index(dihedral.connection_members[3])+1
                 ))
+
+
+def read_lammpsdata(filename, atom_style='full', unit_style='real', potential='lj'):
+    """Read in a lammps data file as a GMSO topology
+
+    Parameters
+    ----------
+    filename : str
+        LAMMPS data file
+    atom_style : str, optional, default='full'
+        Inferred atom style defined by LAMMPS
+    potential: str, optional, default='lj'
+        Potential type defined in data file
+
+    Returns
+    -------
+    top : GMSO Topology
+        A GMSO Topology object
+
+    Notes
+    -----
```

Comment:
This needs some beefing up to clarify what is and is not supported right now. I think, for example, dihedrals aren't read in properly now

Reply:
Add some information as you suggested.  But I think there's room for improvement in future PRs.